### PR TITLE
Add option force update for Single Product and Product List close #25

### LINKED ISSUE #25: Add option for full product update ( Force Update )
### Acceptance Criteria:
- [ ] User can click a button in the edit product page to force aggregate info from Knawat API
- [ ] User can select multiple products and click update from Knawat Button to fetch updated info
- [ ] This button should update all fields including title, description, variations, and images

### PR DESCRIPTION
**Pull request by hamusthekiller**
Description:
Added a part where the user can update a single product in the edit product page using Knawat button for updating.
Also, added a part where the user can select multiple products from the products list in Woocommerce and select Knawat Update from the bulk actions list and the products will be updated.

Steps to test this issue:
1- Download Xampp or any localhost server hosting on your local machine.
2- Download and install Wordpress, Woocommerce, qtranslate-x and all other important plugins that Woocommerce needs to work as expected.
3- Download Knawat Dropshipping plugin and also download recommended plugins by Knawat.
4- Import products into your store.
5- Go to the edit product page in any Knawat product and try to change its values then press update and you should see all these changes reverted to Knawat's original status and updating quantity and price if there is a change in them.
6- Go to all products page and select Knawat Update from bulk actions select box and click apply after that watch the magic happens.
Note: Update Multiple products in the products list might slow down your website speed so don't do it unless 100% sure. Or you can use Knawat dropshipping Plugin to update all products and not slow down Your store.

### CODE REVIEW

File: includes/class-dropshipping-woocommerce-common.php [hunk 1]
```diff
@@ -417,6 +420,42 @@ function knawat_show_notice_for_delete( $post_id ) {
...
+						__('Knawat Dropshipping', 'dropshipping-woocommerce'),
+						array($this, 'knawatproduct_render_metabox'),
+						'product',
+						'side',
+						'high'
+					);
+		}
+	}
+	function knawatproduct_render_metabox(){
+		include KNAWAT_DROPWC_PLUGIN_DIR . 'templates/products/edit_product.php';
```
Comment (ElGreetly) [inline includes/class-dropshipping-woocommerce-common.php]: use include_once()

**Comment by HazemKhaled** [comment]
@hamusthekiller please give a response to the review here

**Comment by hamusthekiller** [comment]
This PR has passed testing. The only PR that didn't pass the one with google tag manager because We need to write shell script for auto update/ deploy

File: templates/products/edit_product.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+    /**
+     * Render Function for updating product in edit page
+     * @return void.
+     */
+    function knawatproduct_metabox(){
```
Comment (HazemKhaled) [inline templates/products/edit_product.php:6]: Codacy found an issue: [knawatproduct_metabox accesses the super-global variable $_GET.](https://app.codacy.com/gh/Knawat/dropshipping-woocommerce/pullRequest?prid=5012389)

File: templates/products/edit_product.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                jQuery("#knawat-update").click(function (e) {
+                    var x = window.location.href;
+                    x += '&knawat-update=true';
+                    document.getElementById('knawat-update').setAttribute('href', x);
+                });
+            </script>
+        </dev>
+<?php
+        if (isset($_GET['knawat-update']) && $_GET['knawat-update'] == true){
+            if (!isset($_GET['product-update'])){
```
Comment (HazemKhaled) [inline templates/products/edit_product.php:22]: Codacy found an issue: [Processing form data without nonce verification.](https://app.codacy.com/gh/Knawat/dropshipping-woocommerce/pullRequest?prid=5012389)

File: templates/products/edit_product.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            <script>
+                jQuery("#knawat-update").click(function (e) {
+                    var x = window.location.href;
+                    x += '&knawat-update=true';
+                    document.getElementById('knawat-update').setAttribute('href', x);
+                });
+            </script>
+        </dev>
+<?php
+        if (isset($_GET['knawat-update']) && $_GET['knawat-update'] == true){
```
Comment (HazemKhaled) [inline templates/products/edit_product.php:21]: Codacy found an issue: [Processing form data without nonce verification.](https://app.codacy.com/gh/Knawat/dropshipping-woocommerce/pullRequest?prid=5012389)

Comment (HazemKhaled) [inline templates/products/edit_product.php:21]: Codacy found an issue: [Direct use of $_GET Superglobal detected.](https://app.codacy.com/gh/Knawat/dropshipping-woocommerce/pullRequest?prid=5012389)

File: templates/products/edit_product.php (hunk 3 above)
Comment (HazemKhaled) [inline templates/products/edit_product.php:22]: Codacy found an issue: [Direct use of $_GET Superglobal detected.](https://app.codacy.com/gh/Knawat/dropshipping-woocommerce/pullRequest?prid=5012389)

File: templates/products/edit_product.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            if (!isset($_GET['product-update'])){
+                Knawat_Dropshipping_Woocommerce_Common:: knawat_dropshipwc_before_single_product();
```
Comment (HazemKhaled) [inline templates/products/edit_product.php:23]: Codacy found an issue: [Avoid using static access to class 'Knawat_Dropshipping_Woocommerce_Common' in method 'knawatproduct_metabox'.](https://app.codacy.com/gh/Knawat/dropshipping-woocommerce/pullRequest?prid=5012389)

**Pull request closed without merging**